src/fn/csv-to-docs.js

```diff
@@ -262,5 +277,6 @@ module.exports = {
   int,
   setCol,
   parseColumn,
-  removeExcludedField
+  removeExcludedField,
+  warnIfDirectoryIsNotEmpty,
```

Comment:
In my opinion, it is a bad pattern to export a function just so that it can be unit tested (though admittedly we do it quite a bit in this codebase.  I just hate to add to it....). It makes it more confusing for the reader to understand what functionality the file is actually offering.  A better approach would be:

- Just test this function as part of the larger workflow by calling `execute`.
- Or, extract this function to a lib file where it could be reused by other files in the future (this functionality seem generic enough that it could be reused for other workflows/actions).  The `lib/sync-fs.js` file seems like a reasonable place to put it. Then, the tests in the `csv-to-docs.spec.js` can mock this functionality and test both cases without caring about the internals of the function...

Reply:
Yes, I referenced some of the older code before doing this.
I'd go with the 2nd option, I was considering it before doing what I did, just wasn't sure where I should be keeping it.

Reply:
I've gone with the 2nd option @jkuester suggested.
However, I've encountered a small `gotcha` . When moving the `warnIfDirectoryIsNotEmpty` function to `sync-fs.js` file, both `warnIfDirectoryIsNotEmpty` and `isDirectoryEmpty`(function used by `warnIfDirectoryIsNotEmpty`) will be located within the same module, so it won't be possible to mock `isDirectoryEmpty` anymore(Reference: https://github.com/sinonjs/sinon/issues/1161#issuecomment-251591197). Hence, when testing `warnIfDirectoryIsNotEmpty`, I will not be mocking `isDirectoryEmpty` anymore and will start mocking `fs.readdirSync`(implementation inside `isDirectoryEmpty`). So, the tests won't exactly be unit tests but kind of a integration test.
Let me know if it's acceptable.

Reply:
:+1: This seems fine to me. I you look for perfection in JS you will always be disappointed!  :sweat_smile: Your proposal seems like the best balance of good structure and good testing!

Reply:
Thanks, @jkuester . Updated.